### PR TITLE
merge autoreload_{inotify,nop}.c into autoreload.c

### PR DESCRIPTION
**Pull request by explosion-mental**
Very simple change. One file less.

### CODE REVIEW

**Comment by explosion-mental** [comment]
taking the oportunity to ask about a 'better' rename to `OPT_DEP_DEFAULT`. Seems to long imo

**Comment by N-R-K** [comment]
This will break make's dependency tracking I'm pretty sure.

```console
$ make HAVE_INOTIFY=0
$ make HAVE_INOTIFY=1  # this won't trigger a rebuild as it should
```

**Comment by explosion-mental** [comment]
hmm on a clean nsxiv build this also doesn't werk

**Comment by N-R-K** [comment]
> hmm on a clean nsxiv build this also doesn't werk

It does work currently on master.

```console
$ make HAVE_INOTIFY=0
CC autoreload_nop.o
c99 -std=c99 -Wall -pedantic -D_XOPEN_SOURCE=700  -DHAVE_LIBGIF=1 -DHAVE_LIBEXIF=0  -DHAVE_LIBWEBP=1 -DHAVE_LIBFONTS=1  -I/usr/include/freetype2 -I/usr/local/include/freetype2 -c -o autoreload_nop.o autoreload_nop.c
CC commands.o
c99 -std=c99 -Wall -pedantic -D_XOPEN_SOURCE=700  -DHAVE_LIBGIF=1 -DHAVE_LIBEXIF=0  -DHAVE_LIBWEBP=1 -DHAVE_LIBFONTS=1  -I/usr/include/freetype2 -I/usr/local/include/freetype2 -c -o commands.o commands.c
CC image.o
c99 -std=c99 -Wall -pedantic -D_XOPEN_SOURCE=700  -DHAVE_LIBGIF=1 -DHAVE_LIBEXIF=0  -DHAVE_LIBWEBP=1 -DHAVE_LIBFONTS=1  -I/usr/include/freetype2 -I/usr/local/include/freetype2 -c -o image.o image.c
CC main.o
c99 -std=c99 -Wall -pedantic -D_XOPEN_SOURCE=700  -DHAVE_LIBGIF=1 -DHAVE_LIBEXIF=0  -DHAVE_LIBWEBP=1 -DHAVE_LIBFONTS=1  -I/usr/include/freetype2 -I/usr/local/include/freetype2 -c -o main.o main.c
GEN version.h
v="$(git describe 2>/dev/null || true)"; \
echo "#define VERSION \"${v:-29}\"" >version.h
CC options.o
c99 -std=c99 -Wall -pedantic -D_XOPEN_SOURCE=700  -DHAVE_LIBGIF=1 -DHAVE_LIBEXIF=0  -DHAVE_LIBWEBP=1 -DHAVE_LIBFONTS=1  -I/usr/include/freetype2 -I/usr/local/include/freetype2 -c -o options.o options.c
CC thumbs.o
c99 -std=c99 -Wall -pedantic -D_XOPEN_SOURCE=700  -DHAVE_LIBGIF=1 -DHAVE_LIBEXIF=0  -DHAVE_LIBWEBP=1 -DHAVE_LIBFONTS=1  -I/usr/include/freetype2 -I/usr/local/include/freetype2 -c -o thumbs.o thumbs.c
CC util.o
c99 -std=c99 -Wall -pedantic -D_XOPEN_SOURCE=700  -DHAVE_LIBGIF=1 -DHAVE_LIBEXIF=0  -DHAVE_LIBWEBP=1 -DHAVE_LIBFONTS=1  -I/usr/include/freetype2 -I/usr/local/include/freetype2 -c -o util.o util.c
CC window.o
c99 -std=c99 -Wall -pedantic -D_XOPEN_SOURCE=700  -DHAVE_LIBGIF=1 -DHAVE_LIBEXIF=0  -DHAVE_LIBWEBP=1 -DHAVE_LIBFONTS=1  -I/usr/include/freetype2 -I/usr/local/include/freetype2 -c -o window.o window.c
LINK nsxiv
c99  -o nsxiv autoreload_nop.o commands.o image.o main.o options.o  thumbs.o util.o window.o  -lImlib2 -lX11   -lgif  -lwebpdemux -lwebp -lXft -lfontconfig
$ make HAVE_INOTIFY=1
CC autoreload_inotify.o
c99 -std=c99 -Wall -pedantic -D_XOPEN_SOURCE=700  -DHAVE_LIBGIF=1 -DHAVE_LIBEXIF=0  -DHAVE_LIBWEBP=1 -DHAVE_LIBFONTS=1  -I/usr/include/freetype2 -I/usr/local/include/freetype2 -c -o autoreload_inotify.o autoreload_inotify.c
LINK nsxiv
c99  -o nsxiv autoreload_inotify.o commands.o image.o main.o options.o  thumbs.o util.o window.o  -lImlib2 -lX11   -lgif  -lwebpdemux -lwebp -lXft -lfontconfig
```

**Comment by explosion-mental** [comment]
Double checked and yep.
Seems to be the only 'dependency' that `make` tracks tho

**Comment by TAAPArthur** [comment]
> It does work currently on master.

Kinda. If you ran  `make HAVE_INOTIFY=0` right afterword nothing would happen. I think anyone relying on changing make args to trigger a rebuild of the necessary files is asking for problems

**Review by TAAPArthur — APPROVED** [review]

File: autoreload.c [hunk 1]
```diff
@@ -109,3 +111,29 @@ bool arl_handle(arl_t *arl)
...
+	(void) filepath;
+}
+
+bool arl_handle(arl_t *arl)
+{
+	(void) arl;
+	return false;
+}
+
+#endif /* INOTIFY */
```
Comment (N-R-K) [inline autoreload.c:139]:
```suggestion
#endif /* HAVE_INOTIFY */
```

**Comment by N-R-K** [comment]
> > It does work currently on master.
> 
> Kinda. If you ran `make HAVE_INOTIFY=0` right afterword nothing would happen.

That's true. Weather having it works sometimes is better than not having it work at all is debateable.

In any case, I'm not a fan of this change, but I won't object to this either since there's some usefulness in reducing the amount of files we currently got laying around.

---

Also just noticed, @explosion-mental you should open PRs from separate branch. Not from `master`, see https://github.com/nsxiv/nsxiv/blob/master/CONTRIBUTING.md#contribution-guideline.

Not a problem this time, but try to keep it in mind next time around.

**Review by N-R-K — APPROVED** [review]
Had a change of heart. Probably best for it to *consistently* not work rather than having it *sometimes* work.

**Comment by explosion-mental** [comment]
> you should open PRs from separate branch. Not from master, see https://github.com/nsxiv/nsxiv/blob/master/CONTRIBUTING.md#contribution-guideline.

whoops

**Pull request merged**